### DISCUSSION TITLE
Unable to reference libraries on an iASP

### DISCUSSION THREAD

**Discussion by StuBramley** [codefori/vscode-ibmi#2322] [Q&A]
Code for IBM i version 2.13.5 - clean installation on new installation of VS Code 1.94.2 

Steps to reproduce:
- Create new connection
- Add a library on an iASP to the library list
Or
- Create a filter in the object browser using a library on an iASP 

Error message 'Library <library> does not exist' will be displayed. 

I have tried setting the 'Source ASP' setting on the 'Source Code' settings tab but this does not have any impact. I do have access to QSYS2.ASP_INFO. 

There does not appear to be any other way to try and specify an iASP on the connection properties.

Please note - developers on our site do not have *SECADM rights and cannot change their profile's job description so cannot set the iASP with this method.

**Comment by SJLennon** [comment]
@StuBramley 
You should take a look at the "[Help and Support](https://codefori.github.io/docs/help-and-support/)" section of the documentation. There are a couple of things you can do to help the maintainers.

**Comment by StuBramley** [comment]
> @StuBramley You should take a look at the "[Help and Support](https://codefori.github.io/docs/help-and-support/)" section of the documentation. There are a couple of things you can do to help the maintainers.

I tried but can't use the 'Report an Issue' tool unfortunately.. the software is running in a Citrix environment and the server cannot access the internet. Happy to try and gather any information manually that will help..

**Comment by worksofliam** [comment]
Sadly it's been this way for a little while. For now, the only way to make VS Code understand iASP is to set it at the user profile level.
